velox/exec/tests/TableWriteTest.cpp

```diff
@@ -2004,20 +2057,25 @@ TEST_P(
       numTableWriterCount_,
       connector::hive::LocationHandle::TableType::kExisting);
 
-  VELOX_ASSERT_THROW(
-      AssertQueryBuilder(plan)
-          .connectorConfig(
-              kHiveConnectorId, HiveConfig::kImmutablePartitions, "true")
-          .copyResults(pool()),
-      "Unpartitioned Hive tables are immutable.");
+  if (testParam_.hiveImmutablePartitions()) {
+    VELOX_ASSERT_THROW(
+        AssertQueryBuilder(plan).copyResults(pool()),
+        "Unpartitioned Hive tables are immutable.");
+  } else {
+    AssertQueryBuilder(plan).copyResults(pool());
+  }
 }
 
 TEST_P(UnpartitionedTableWriterTest, appendToAnExistingUnpartitionedTable) {
-  // This test uses the default value "false" for the "immutable_partitions"
-  // config allowing writes to an existing unpartitioned table.
+  // This test uses the default value "false" for the
```

Comment:
I have tried but this is very tricky because our test extends from HiveConnectorTestBase, and when HiveConnectorTestBase is setup, it would finish initializing all the connectors. 

So we have to set the config for hive connector in our test class' constructor and can't change dynamically in the test function.

Reply:
We can have a default setup but connector provides unregisterConnector? Thanks!